app/category/addCategory.js

```diff
@@ -0,0 +1,79 @@
+const Joi = require('joi');
+const { validator } = require('../util/validator');
+const { sendError } = require('../util/error.js');
+
+exports.addCategory_v1 = (req, res, next) => {
+    validateAddCategoryRequest(req)
+    .then((payload) => isCategoryExist(payload))
+    .then((payload) => addCategory(payload))
+    .then((category) => { res.json({ 
+        code: 200, 
+        'id': category.insertedId,
+        'categoryName': req.body.categoryName
+    }) }).catch((errorCode) => {
+        switch(errorCode) {
+            case 422: return sendError(res, errorCode, 'Invalid request!');
+            case 409: return sendError(res, errorCode, 'Category exist!');
+            case 504: return sendError(res, errorCode, 'Unable to fetch categories from store!');
+            default: return sendError(res, 500, 'Something went wrong!');
+        }
+    })
+}
+
+function validateAddCategoryRequest(request) {
```

Comment:
❓ Shouldn't this also check if the user is a `admin`/`shop-owner`?

Reply:
Hey, we have not added role based permission model so far. So it will allow all type of users. However, we have a story where we will onboard user roles and additional security layer to check whether user is allowed to call this api or not. 

This additional sec layer will be at auth level i.e. route file. So basically it when backend receives a request it will check token, user validity and will also check user permission and then only will redirect user to this function.